### ISSUE TITLE
[Paper Note] Guiding LLM Post-training Data Engineering with Model Internals from Sparse Autoencoders, Yi Jing+, arXiv'26, 2026.05

### ISSUE THREAD

**Issue by AkihikoWatanabe** [AkihikoWatanabe/paper_notes#5626]
# URL
- https://arxiv.org/abs/2605.27354
# Authors
  - Yi Jing
  - Zao Dai
  - Jinwu Hu
  - Zijun Yao
  - Lei Hou
  - Juanzi Li
  - Xiaozhi Wang
# Abstract
  - Model internals encode rich information about how a large language model (LLM) processes its training data; however, post-training data engineering largely relies on external signals and ignores rich intrinsic signals lying in model internals. We propose SAERL, a data engineering framework for LLM reinforcement learning (RL). It models three intrinsic data properties: diversity, difficulty, and quality, using model internals extracted with Sparse Autoencoder (SAE), an advanced mechanistic interpretability tool. Each property grounds a concrete data engineering operation: SAE-space clustering with moderate batch mixing for batch diversity control, a difficulty proxy for easy-to-hard curriculum ordering, and a quality probe for data filtering. SAERL improves average accuracy by 3.00% over vanilla GRPO and reaches target accuracy with 20% fewer training steps on Qwen2.5-Math-1.5B, with consistent gains across model scales and RL algorithms. Experiments show that SAE transfers effectively across model families and scales, serving as a lightweight and reusable data engineering tool. These results demonstrate that model internals are a powerful and practical source of signals for post-training data engineering.
# Translation (by gpt-5-nano)
- 本論では、モデル内部情報が大規模言語モデル（LLM）が訓練データを処理する方法に関する豊富な情報を符号化している一方で、事後のデータエンジニアリングは主に外部信号に依存し、モデル内部に潜む豊かな内在信号を無視していることを指摘する。SAERL（LLM の強化学習のためのデータエンジニアリングフレームワーク）を提案する。SAERL は、Sparse Autoencoder（SAE、先進の機構的解釈性ツール）を用いて抽出されたモデル内部情報を用い、三つの内在的データ特性（多様性、難易度、品質）をモデル化する。各特性は具体的なデータエンジニアリング操作を支える：適度なバッチ混合を伴う SAE-space クラスタリングによるバッチ多様性の制御、易から難へのカリキュラム順序付けのための難易度代理指標、データフィルタリングのための品質プローブ。SAERL は vanilla GRPO に対して平均精度を 3.00% 向上させ、Qwen2.5-Math-1.5B で 20% 少ないトレーニングステップで目標精度に到達し、モデルのスケールと RL アルゴリズムを問わない一貫した利得を得る。実験は SAE がモデルファミリとスケールを越えて効果的に転移することを示し、軽量で再利用可能なデータエンジニアリングツールとして機能する。これらの結果は、モデル内部がポスト訓練データエンジニアリングの強力で実用的な信号源になり得ることを示している。
# Summary (by gpt-4o-mini)
- モデル内部情報がLLMのデータ処理方法に重要である一方、外部信号に依存したデータエンジニアリングは内在信号を無視していることを指摘。SAERLを提案し、Sparse Autoencoderを用いて多様性、難易度、品質の三つのデータ特性をモデル化。これにより、バッチ多様性や難易度の順序づけ、データフィルタリングを実現。SAERLは平均精度を3.00%向上させ、少ないトレーニングステップで目標精度に達することを示し、効果的なデータエンジニアリングツールとしての役割を果たすことが確認された。

**Comment by AkihikoWatanabe** [comment]
元ポスト:https://x.com/yi_jing04/status/2059595634060230748?s=46&t=Y6UuIHB0Lv0IpmFAjlc2-Q

**Comment by AkihikoWatanabe** [comment]
SAEのrepresentationを、interpretabilityに活用するのではなく、post-trainingの学習データに対するdata engineeringに使うことで、costのかかる手法ではなく**より低コストで**data engineeringを実現したい、という気持ちの研究。提案手法では、SAEによって獲得されるrepresentationに基づいてpost-trainingの学習データに対して、
- 多様性: SAErepresentationを用いてクラスタリングを実施し活用
- 難易度: 軽量なElasticNetに基づく回帰モデル（特徴量はSAE representation）によって難易度予測モデルを学習し、クラスタIDに基づいて難易度をキャリブレーション
- 品質: SAE representationに基づいてqualityを判断する二値分類器を学習しその確率値を使うようである

ぱっと見よくわからないのが、
- difficulty-labeledなsubsetの正体はなんなのか？
- それは幅広いドメインで入手可能なものなのか？
- in-distributionな難易度であればElasticNetで予測できたということだが、in-distributionなdifficulty-labeledなデータがないと提案手法は原則として適用できないということなのか？

という疑問はある。


<img width="1135" height="675" alt="Image" src="https://github.com/user-attachments/assets/4448abbe-0881-4762-b0f6-5a8b5cbe7ea9" />
